docs/guide.md

```diff
@@ -0,0 +1,69 @@
+
+# Guide to the Dafny Core Library
+
+_THIS DOCUMENT IS A DRAFT UNDER ACTIVE DEBATE AND EDITING_
+
+A programming language library encapsulates a useful set of basic
+functionality that programmers can then incorporate in their custom
+program without having to rewrite, debug, or verify.
+
+Like other languages, the Dafny programming language and tool set includes
+libraries that provide functionality in various domains. This document describes 
+the _Dafny Core Library_. Dafny programs are copmponed of modules; to ensure
+verification soundness, Dafny imposes a strict non-circular dependency relationship
+among modules. This Core library contains functionality that may be needed
+by other libraries, but it itself is standalone.
+
+All the supported, delivered libraries will
+- be stable --- a minimum of backwards incompatible changes
+- be verified --- proved to meet specifications and tested with expected use cases,
+with each release of the library and of the Dafny toolset
+- be efficient --- efficient in both proofs and executables that use the libraries
```

Comment:
We should also have something about executable code being compilable for every supported target language by default, unless otherwise documented (such as the current state of FileIO)

Reply:
Added a bullet stating this